webknossos/webknossos/geometry/vec3_int.py

```diff
@@ -101,104 +63,48 @@ def with_y(self, new_y: int) -> "Vec3Int":
-    # which is a different behavior in the superclass! Hence the type:ignore.
-    def __mul__(self, other: Union[int, "Vec3IntLike"]) -> "Vec3Int":  # type: ignore[override]
-        return self._element_wise(other, mul)
-
-    def __floordiv__(self, other: Union[int, "Vec3IntLike"]) -> "Vec3Int":
-        return self._element_wise(other, floordiv)
-
-    def __mod__(self, other: Union[int, "Vec3IntLike"]) -> "Vec3Int":
-        return self._element_wise(other, mod)
-
-    def __neg__(self) -> "Vec3Int":
-        return Vec3Int.from_xyz(-self.x, -self.y, -self.z)
-
-    def ceildiv(self, other: Union[int, "Vec3IntLike"]) -> "Vec3Int":
-        return (self + other - 1) // other
-
-    def pairmax(self, other: Union[int, "Vec3IntLike"]) -> "Vec3Int":
-        return self._element_wise(other, max)
+    @staticmethod
+    def from_xyz(x: int, y: int, z: int) -> "Vec3Int":
+        """Use Vec3Int.from_xyz for fast construction."""
 
-    def pairmin(self, other: Union[int, "Vec3IntLike"]) -> "Vec3Int":
-        return self._element_wise(other, min)
+        # By calling __new__ of tuple directly, we circumvent
+        # the tolerant (and potentially) slow Vec3Int.__new__ method.
+        return tuple.__new__(Vec3Int, (x, y, z))
 
-    def prod(self) -> int:
-        return self.x * self.y * self.z
+    @staticmethod
+    def from_vec3_float(vec: Tuple[float, float, float]) -> "Vec3Int":
+        return Vec3Int(int(vec[0]), int(vec[1]), int(vec[2]))
 
-    def __repr__(self) -> str:
-        return f"Vec3Int({self.x},{self.y},{self.z})"
+    @staticmethod
+    def from_vec_or_int(vec_or_int: Union["Vec3IntLike", int]) -> "Vec3Int":
+        if isinstance(vec_or_int, int):
+            return Vec3Int.full(vec_or_int)
 
-    def add_or_none(self, other: Optional["Vec3Int"]) -> Optional["Vec3Int"]:
-        return None if other is None else self + other
+        return Vec3Int(vec_or_int)
 
-    def moveaxis(
-        self, source: Union[int, List[int]], target: Union[int, List[int]]
-    ) -> "Vec3Int":
-        """
-        Allows to move one element at index `source` to another index `target`. Similar to
-        np.moveaxis, this is *not* a swap operation but instead it moves the specified
-        source so that the other elements move when necessary.
-        """
+    @staticmethod
+    def from_str(string: str) -> "Vec3Int":
+        if re.match(r"\(\d+,\d+,\d+\)", string):
+            return Vec3Int(tuple(map(int, re.findall(r"\d+", string))))
 
-        # Piggy-back on np.moveaxis by creating an auxiliary array where the indices 0, 1 and
-        # 2 appear in the shape.
-        indices = np.moveaxis(np.zeros((0, 1, 2)), source, target).shape
-        arr = self.to_np()[np.array(indices)]
-        return Vec3Int(arr)
+        return Vec3Int.full(int(string))
 
     @classmethod
-    def zeros(cls) -> "Vec3Int":
+    def zeros(cls, length: int = 3) -> "Vec3Int":
+        del length
```

Comment:
You are right the length parameter is not needed and therefore is deleted right after calling the function. It is present to have the same method signature as in the base class, to avoid the lint warning https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/arguments-differ.html

Reply:
So del is only used to satisfy the linter (e.g. Unused Argument Warning)? In that case you could add an underscore the argument name to mark it as unused and satisfy the linter, no? e.g. (..., _length=3).

Reply:
Done.